audacity/audacity
# 6618 audiocom redirect tagging

Resolves: #6618




Recommended:

## QA

When redirected to audio.com, the actions below should lead to the URL opened in the browser ending with
`mtm_campaign=Audacity&mtm_source=<audacity version>&mtm_content=<see below>`,
e.g. `https://audio.com/matthieu-hodgkinson/audio/french-break-136-bpm-pl/edit?mtm_campaign=Audacity&mtm_source=Audacity-3.6.0-alpha-20240610&mtm_content=Share_Audio_Button`
Note that this should also apply to the "link-account" button associated with these actions, as well as to the 
"Visit audio.com" button in case the max number of projects was exceeded.

### Example
I want to save my project to the cloud,
![image](https://github.com/audacity/audacity/assets/22740106/23c5b092-d1ae-4a77-9812-ca9441006b13)

but there have to link my account: 
![image](https://github.com/audacity/audacity/assets/22740106/defcf838-19fc-4fdd-ba04-406e10ffab07)

It leads me to audio.com and the link has `Save_to_Cloud_Menu` appended:
![image](https://github.com/audacity/audacity/assets/22740106/e6228408-f009-4aa0-b994-d9ced2ccd649)

Once I have linked my account, upload begins and then I see the success dialog, and clicking on "view my project" leads me to 
![image](https://github.com/audacity/audacity/assets/22740106/67b0871d-2b6c-4744-a7a1-4fd73c8a17ab), which also has that `Save_to_Cloud_Menu` appended.

I save another project, and say I reached my limit. I see 
![image](https://github.com/audacity/audacity/assets/22740106/275b5cef-9fea-423e-b2e9-ea7d23e54209)
and clicking "Visit audio.com" leads me to my projects page, 
![image](https://github.com/audacity/audacity/assets/22740106/6c405916-d254-4ae4-b7ab-033c4b1da5d9), and the link also has `Save_to_Cloud_Menu` appended.

### Actions (to be checked by QA)

| Action                            | string                            | notes |
| --------------------------------- | --------------------------------- |---|
|![image](https://github.com/audacity/audacity/assets/22740106/1bdcb768-953c-4f93-91d7-3d0029e91d44) | `Share_Audio_Button`                | |
| ![image](https://github.com/audacity/audacity/assets/22740106/50fb33fd-97d9-44f9-b945-d2c780bcd97a)                  | `Share_Audio_Menu`                  | |
| ![image](https://github.com/audacity/audacity/assets/22740106/06fdf9cf-5c82-4302-8bee-e968d0158b1a)           | `Share_Audio_Export_Menu`           | |
| ![image](https://github.com/audacity/audacity/assets/22740106/4b083114-6579-402e-ac88-8dfbb1953767)| `Share_Audio_Export_Extra_Menu`     | mp3 or any other format but not "selected audio" |
| ![image](https://github.com/audacity/audacity/assets/22740106/b33eda05-b6b3-46c4-be3c-09a6ffdb3c88)               | `Save_to_Cloud_Menu`                | |
| ![image](https://github.com/audacity/audacity/assets/22740106/86389bf8-2802-4ea2-bc18-9d958f9e1de2)  | `Save_Project_Save_to_Cloud_Menu`   | and then when the modal opens, use the cloud option |
| ![image](https://github.com/audacity/audacity/assets/22740106/b0a2f206-22f8-450c-9f20-dc0d7311b8e4)                       | `Prefs_Panel`                       |  |
| ![image](https://github.com/audacity/audacity/assets/22740106/97e2b751-5b4d-439c-a121-921cf893a3a7)   | `Update_Cloud_Audio_Preview_Menu`   | |
| ![image](https://github.com/audacity/audacity/assets/22740106/5888556e-bc33-482c-8253-8515fc9a878d)   | `Link_Audiocom_Account_Help_Menu`   | If done successfully, there shouldn't be a redirect. If it fails, though, then there should, and the string should be in the URL. One way of failing it is to give some random token. (There are several other reasons why this might fail but testing them all would probably be an overkill. |
| ![image](https://github.com/audacity/audacity/assets/22740106/176f23ee-6335-4294-bbc6-1d98debaf9cb)              | Open_From_Cloud_Menu              | |
